src/main/java/com/shaft/database/DBActions.java

```diff
@@ -0,0 +1,190 @@
+			ReportManager.attachAsStep("Passed", "DB Connection Log", log);
+		}
+	}
+
+	private void passAction(String actionName, String testData) {
+		passAction(actionName, testData, null);
+	}
+
+	private void failAction(String actionName, String testData, String log) {
+		String message = "Failed to perform action [" + actionName + "].";
+		if (testData != null) {
+			message = message + " With the following test data [" + testData + "].";
+		}
+		ReportManager.log(message);
+		if (log != null) {
+			ReportManager.attachAsStep("Failed", "DB Connection Log", log);
+		}
+		Assert.fail(message);
+	}
+
+	private void failAction(String actionName, String testData) {
+		failAction(actionName, testData, null);
+	}
+
+	/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	//////////////////////////////////// [Public] Core Database Actions
+	/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	/**
+	 * Setup connection to database
+	 * 
+	 * @throws ClassNotFoundException
+	 * @throws SQLException
+	 * 
+	 * 
+	 */
+	
+	private void createConnection() {
+		String connectionString = "";
+		try {
+			switch (dbType.toLowerCase().trim()) {
+
+			case ("mysql"):
+				connectionString = "jdbc:mysql://" + dbServerIP + ":" + dbPort + "/" + dbName;
+				break;
+
+			case ("sqlserver"):
+				connectionString = "jdbc:sqlserver://" + dbServerIP + ":" + dbPort + ";databaseName=" + dbName;
+				break;
+
+			case ("postgresql"):
+				connectionString = "jdbc:postgresql://" + dbServerIP + ":" + dbPort + "/" + dbName;
+				break;
+
+			default:
+				ReportManager.log("Database not supported");
+				failAction("createConnection", dbType);
+				break;
+			}
+			connection = DriverManager.getConnection(connectionString, username, password);
+			passAction("createConnection", connectionString);
+		} catch (SQLException e) {
+			ReportManager.log(e);
+			failAction("createConnection", connectionString);
+		}
+	}
+	
+	/**
+	 * prepare statement for database connection to perform execute query
+	 * 
+	 * 
+	 */
+	
+	private void createStatement() {
+		try {
+			statement = connection.createStatement();
+			passAction("createStatement", connection.toString());
+		} catch (SQLException e) {
+			ReportManager.log(e);
+			failAction("createStatement", connection.toString());
+		}
+	}
+
+	/**
+	 * execute DB query and return the result as ResultSet object
+	 * 
+	 * @param dbQuery
+	 * @return ResultSet
+	 * @throws SQLException
+	 * 
+	 * return ResultSet
+	 */
+	public ResultSet executeSelectQuery(String dbQuery) {
+		ResultSet resultSet = null;
+		try {
+			createConnection();
+			createStatement();
+			resultSet = statement.executeQuery(dbQuery);
+			passAction("executeSelectQuery", dbQuery);
+		} catch (SQLException e) {
+			ReportManager.log(e);
+			failAction("executeSelectQuery", dbQuery);
+		}
+		return resultSet;
```

Comment:
as discussed, it is a must to implement one of those two options here:
1. return a string value so that the assertions library could be used.
2. provide more methods to process the ResultSet object and extract string data from it, so that the assertions library could be used.

Reply:
let's discuss this in the meeting isA

Reply:
as agreed, I will develop this part after merging your contribution to a separate feature branch.